### DISCUSSION TITLE
Enable encryption for existing OSDs

### DISCUSSION THREAD

**Discussion by xoxys** [rook/rook#12358] [Q&A] [answered]
Is there any way to enable encryption with `encryptedDevice: "true"` in an existing cluster, or do I have to start from scratch?

**Comment by xoxys** [comment]
Can I just follow the [steps to replace OSDs](https://rook.io/docs/rook/v1.11/Storage-Configuration/Advanced/ceph-osd-mgmt/#replace-an-osd) to replace OSD by OSD to get new ones with encryption enabled?

**Reply by travisn** [reply, marked as answer]
Yes, after you enable encryption, new OSDs would be created as encrypted. So if you replace the OSDs one-by-one and wait for the ceph status to be healthy between each one, this would work.

**Reply by xoxys** [reply]
Awesome, thanks for the confirmation :rocket:

**Reply by xoxys** [reply]
Thank you for the great documentation. The replacement of the OSD's worked flawlessly.
